src/utils.jl

```diff
@@ -178,12 +178,13 @@ end
 function getsource(source, use_mmap)
     if source isa Vector{UInt8}
         return source
-    elseif source isa Cmd
-        return Base.read(source)
-    elseif use_mmap && !isa(source, IO)
-        return Mmap.mmap(source)
+    elseif source isa Cmd || (source isa AbstractPath && !isa(source, SystemPath))
+        return Base.read(open(source))
     elseif !isa(source, IO)
-        m = Mmap.mmap(source)
+        m = Mmap.mmap(source isa AbstractPath ? open(source) : source)
```

Comment:
I think this is effectively the same as
```suggestion
        m = open(Mmap.mmap, source)
```
since all `Mmap.mmap(source::String)` does is basically `open(Mmap.mmap, source)` anyway